### PR TITLE
(DOCSP-30625): Add docs release notes for w/e June 16

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-30625

### CODE REVIEW

File: docs-release-notes.md [hunk 1]
```diff
@@ -1,10 +1,18 @@
+# June 16, 2023
+
+## Device Sync
+
+- App Builder/Tiered Device Sync: Clarify the opening Tip about starting a PoC with a Template App to get an App pre-configured for Device Sync
```
Comment (lindseymoore) [inline docs-release-notes.md:5]: Having trouble finding where this is in the TOC (don't see a 'Tiered Device Sync' page under App Builder's Resources)

**Review by lindseymoore — CHANGES_REQUESTED** [review]
Just can't find one of the pages in the TOC. Other than that, LGTM!

File: docs-release-notes.md (hunk 1 above)
Comment (dacharyc) [reply docs-release-notes.md:5]: Yeah, this is (intentionally) a private page currently, so it does not appear in the ToC. Well spotted, though!

**Review by lindseymoore — APPROVED** [review]
Page is private, LGTM!

**Pull request merged**